README.md

```diff
@@ -7,230 +7,205 @@
+complete OP-TEE repo. In this case, the necessary OP-TEE libraries are
+initialized in the setup process.
 
-Before building examples, the environment should be properly setup.
+The complete list of prerequisites can be found here: [OP-TEE
+Prerequisites](https://optee.readthedocs.io/en/latest/building/prerequisites.html).
 
 ``` sh
-$ source environment
-```
-
-By default, the target platform is `aarch64`. If you want to build for the `arm`
-target, you can setup `ARCH` before `source environment`:
-
-```sh
-$ export ARCH=arm
-$ source environment
+# install dependencies
+sudo apt-get install android-tools-adb android-tools-fastboot autoconf \
+  automake bc bison build-essential ccache cscope curl device-tree-compiler \
+  expect flex ftp-upload gdisk iasl libattr1-dev libc6:i386 libcap-dev \
+  libfdt-dev libftdi-dev libglib2.0-dev libhidapi-dev libncurses5-dev \
+  libpixman-1-dev libssl-dev libstdc++6:i386 libtool libz1:i386 make \
+  mtools netcat python-crypto python3-crypto python-pyelftools \
+  python3-pycryptodome python3-pyelftools python-serial python3-serial \
+  rsync unzip uuid-dev xdg-utils xterm xz-utils zlib1g-dev
 ```
 
-#### 4. Build Rust applications
-
-Before building built-in rust examples, you need to build OP-TEE libraries using:
-
-``` sh
-$ make optee
-```
+Alternatively, you can use a docker container built with our
+[Dockerfile](Dockerfile).
 
-Run this command to build all Rust examples:
+After installing dependencies or building the Docker image, fetch the source code from the official GitHub repository:
 
 ``` sh
-$ make examples
+# clone the project
+git clone https://github.com/apache/incubator-teaclave-trustzone-sdk.git
+cd incubator-teaclave-trustzone-sdk
 ```
 
-Or build your own CA and TA:
+### Build & Install
 
-```
-$ make -C examples/[YOUR_APPLICATION]
-```
+To build the project, the Rust environment and several related submodules are required.
 
-#### 5. Run Rust applications
+If you are working without QEMUv8, by default, the `OPTEE_DIR` is
+  `incubator-teaclave-trustzone-sdk/optee/`. OP-TEE submodules (`optee_os`,
+`optee_client` and `build`) will be initialized automatically in `setup.sh`.
 
-The shared folder is needed to share CAs and TAs with the QEMU guest system.
-Recompile QEMU in OP-TEE to enable QEMU virtfs:
+If you are building within QEMUv8 or already have the [OP-TEE
+repository](https://github.com/OP-TEE)  cloned somewhere, you can set the OP-TEE
+root directory with:
 
-```
-$ (cd $OPTEE_DIR/build && make QEMU_VIRTFS_ENABLE=y qemu)
+```sh
+export OPTEE_DIR=[YOUR_OPTEE_DIR]
 ```
 
-Note: the path `/project/root/dir/` should be replaced as the root directory of
-your local project "Teaclave TrustZone SDK". Copy all the Rust examples or your
-own applications to the shared folder:
+Note: your OPTEE root directory should have `build/`, `optee_os/` and
+`optee_client/` as sub-directory.
+
+Run the script as follows to install the Rust environment and initialize submodules:
 
```

Comment:
Add some bullets or numbers to distinguish the different steps, because the notes inside steps may be confusing.

e.g.
1. Set the OP-TEE directory
    ...
2. Install the Rust environment and initialize submodules
    ...
3. Build OP-TEE libraries